Compilation warning on 2.3.3

## Expected Behavior
No warning should appear during compilation

## Current Behavior
```
In file included from /usr/include/qt/QtGui/qtguiglobal.h:43,
                 from /usr/include/qt/QtWidgets/qtwidgetsglobal.h:43,
                 from /usr/include/qt/QtWidgets/qapplication.h:43,
                 from /usr/include/qt/QtWidgets/QApplication:1,
                 from /build/keepassxc/src/keepassxc-2.3.3/src/gui/Application.h:23,
                 from /build/keepassxc/src/keepassxc-2.3.3/src/gui/Application.cpp:20:
/build/keepassxc/src/keepassxc-2.3.3/src/gui/Application.cpp: In static member function ‘static void Application::handleUnixSignal(int)’:
/build/keepassxc/src/keepassxc-2.3.3/src/gui/Application.cpp:240:29: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Wunused-result]
             Q_UNUSED(::write(unixSignalSocket[0], &buf, sizeof(buf)));
                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/keepassxc/src/keepassxc-2.3.3/src/gui/Application.cpp: In member function ‘void Application::quitBySignal()’:
/build/keepassxc/src/keepassxc-2.3.3/src/gui/Application.cpp:252:20: warning: ignoring return value of ‘ssize_t read(int, void*, size_t)’, declared with attribute warn_unused_result [-Wunused-result]
     Q_UNUSED(::read(unixSignalSocket[1], &buf, sizeof(buf)));
              ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

## Steps to Reproduce (for bugs)
1. Compile 2.3.3 (with gcc 8.1.0)

## Context
Compiling 2.3.3 for Arch Linux package.

--- a/src/gui/Application.cpp
+++ b/src/gui/Application.cpp
@@ -235,7 +235,7 @@ void Application::handleUnixSignal(int sig)
     case SIGINT:
     case SIGTERM: {
         char buf = 0;
-        Q_UNUSED(::write(unixSignalSocket[0], &buf, sizeof(buf)));
+        Q_UNUSED(!::write(unixSignalSocket[0], &buf, sizeof(buf)));
         return;
     }
     case SIGHUP:
@@ -247,7 +247,7 @@ void Application::quitBySignal()
 {
     m_unixSignalNotifier->setEnabled(false);
     char buf;
-    Q_UNUSED(::read(unixSignalSocket[1], &buf, sizeof(buf)));
+    Q_UNUSED(!::read(unixSignalSocket[1], &buf, sizeof(buf)));
     emit quitSignalReceived();
 }
 #endif

--- a/tests/TestKeys.cpp
+++ b/tests/TestKeys.cpp
@@ -235,7 +235,7 @@ void TestKeys::benchmarkTransformKey()
 
     QBENCHMARK
     {
-        Q_UNUSED(compositeKey->transform(kdf, result));
+        Q_UNUSED(!compositeKey->transform(kdf, result));
     };
 }
 

--- a/tests/gui/TestGui.cpp
+++ b/tests/gui/TestGui.cpp
@@ -853,7 +853,7 @@ void TestGui::testTotp()
 void TestGui::testSearch()
 {
     // Add canned entries for consistent testing
-    Q_UNUSED(addCannedEntries());
+    addCannedEntries();
 
     auto* toolBar = m_mainWindow->findChild<QToolBar*>("toolBar");
 
@@ -1007,7 +1007,7 @@ void TestGui::testSearch()
 void TestGui::testDeleteEntry()
 {
     // Add canned entries for consistent testing
-    Q_UNUSED(addCannedEntries());
+    addCannedEntries();
 
     auto* groupView = m_dbWidget->findChild<GroupView*>("groupView");
     auto* entryView = m_dbWidget->findChild<EntryView*>("entryView");
@@ -1673,10 +1673,8 @@ void TestGui::testAutoType()
     entryView->selectionModel()->clearSelection();
 }
 
-int TestGui::addCannedEntries()
+void TestGui::addCannedEntries()
 {
-    int entries_added = 0;
-
     // Find buttons
     auto* toolBar = m_mainWindow->findChild<QToolBar*>("toolBar");
     QWidget* entryNewWidget = toolBar->widgetForAction(m_mainWindow->findChild<QAction*>("actionEntryNew"));
@@ -1689,22 +1687,17 @@ int TestGui::addCannedEntries()
     QTest::keyClicks(titleEdit, "test");
     auto* editEntryWidgetButtonBox = editEntryWidget->findChild<QDialogButtonBox*>("buttonBox");
     QTest::mouseClick(editEntryWidgetButtonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
-    ++entries_added;
 
     // Add entry "something 2"
     QTest::mouseClick(entryNewWidget, Qt::LeftButton);
     QTest::keyClicks(titleEdit, "something 2");
     QTest::keyClicks(passwordEdit, "something 2");
     QTest::mouseClick(editEntryWidgetButtonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
-    ++entries_added;
 
     // Add entry "something 3"
     QTest::mouseClick(entryNewWidget, Qt::LeftButton);
     QTest::keyClicks(titleEdit, "something 3");
     QTest::mouseClick(editEntryWidgetButtonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
-    ++entries_added;
-
-    return entries_added;
 }
 
 void TestGui::checkDatabase(QString dbFileName)

--- a/tests/gui/TestGui.h
+++ b/tests/gui/TestGui.h
@@ -72,7 +72,7 @@ private slots:
     void testTrayRestoreHide();
 
 private:
-    int addCannedEntries();
+    void addCannedEntries();
     void checkDatabase(QString dbFileName = "");
     void triggerAction(const QString& name);
     void dragAndDropGroup(const QModelIndex& sourceIndex,

